blog/2021-08-12-state_class_total.md

```diff
@@ -0,0 +1,85 @@
+---
+author: Erik Montnémery
+authorURL: https://github.com/emontnemery
+title: "New sensor state classes: total and total_increasing"
+---
+
+Two new state classes, `STATE_CLASS_TOTAL` and `STATE_CLASS_TOTAL_INCREASING` have been
+added. In addition, it is no longer mandatory to set the `last_reset` attribute for
+`sum` statistics to be generated. The driver for the changes is to make it easier to
+integrate with devices, like utility meters.
+
+### State classes
+
+There are now 3 defined state classes:
+
+- `STATE_CLASS_MEASUREMENT`, the state represents a measurement in present time, for 
+   example a temperature, electric power, etc. For supported sensors, statistics of 
+   hourly min, max and average sensor readings are compiled.
+- `STATE_CLASS_TOTAL`, the state represents a total amount that can both increase and
+   decrease, e.g. the value of a stock portfolio. When supported, the accumulated growth
+   or decline of the sensor's value since it was first added is updated hourly.
+- `STATE_CLASS_TOTAL_INCREASING`, a monotonically increasing total, e.g. an amount of
+   consumed gas, water or energy. When supported, the accumulated growth of the sensor's
+   value since it was first added is updated hourly.
+
+#### `STATE_CLASS_MEASUREMENT`
+
+For sensors with state_class `STATE_CLASS_MEASUREMENT`, it is deprecated to set the
+`last_reset` attribute, and it will be ignored in Home Assistant 2021.10.
+
+#### `STATE_CLASS_TOTAL`
+
+For sensors with state_class `STATE_CLASS_TOTAL`, the `last_reset` attribute can
+optionally be set to gain manual control of meter cycles; each time last_reset changes
+the corresponding value is used as the zero-point when calculating `sum` statistics.
+If last_reset is not set, the sensor's value when it was first added is used as the
+zero-point when calculating `sum` statistics.
+
+Example of `STATE_CLASS_TOTAL` without last_reset:
+
+| t    | state  | sum    |
+| ---: | -----: | -----: |
+|   0  |  1000  |     0  |
+|   1  |  1010  |    10  |
+|   2  |     0  | -1000  |
```

Comment:
```suggestion
|   2  |     0  | -1010  |
```
?

Reply:
I think it should in fact be -990; the zero-point is 1000, so the sensor's state up to t=2 is: (1000-1000) + (1010-1000) + (0 - 1000) = -990

Reply:
I think it's correct as is:
- At t=2, the sensor's value is (1010-1000) + (0 - 1010) = -1000
- At t=3, the sensor's value is (1010-1000) + (0 - 1010) + (5 - 0) = -995